src/main/java/org/hyperskill/hstest/testing/execution/searcher/CppSearcher.java

```diff
@@ -0,0 +1,24 @@
+package org.hyperskill.hstest.testing.execution.searcher;
+
+import org.hyperskill.hstest.testing.execution.runnable.RunnableFile;
+
+/**
+ * Searches for C++ runnable files
+ * (files with main function)
+ * in the given directory
+ * and returns the first one found.
+ */
+public class CppSearcher extends BaseSearcher {
+    @Override
+    public String extension() {
+        return ".cpp";
+    }
+
+    @Override
+    public RunnableFile search(String whereToSearch) {
+        return simpleSearch(whereToSearch,
+                "int main()",
+                "(^|\n) *int +main\\(.*\\)"
```

Comment:
I suggest 
1. `main *\\(` (arbitrary empty spaces between the `main` and `(`
2. Instead of `.*` - `[^)]*` - any symbols except the `)` before the actual `)` (so there should the the only one `)`

Reply:
Thank you, this is not so strict in relation to the code style.

Reply:
Well, I guess you want to execute users's code regardless of the code style quality of this code